File: smoke_test.py
import asyncio
from playwright.async_api import async_playwright

async def run():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        routes = ['/', '/apply', '/auth', '/redesign', '/track']
        base_url = 'http://localhost:8080'

        for route in routes:
            print(f"\n--- Testing route: {route} ---")
            errors = []
            logs = []

            page.on("pageerror", lambda err: errors.append(f"Page Error: {err}"))
            page.on("console", lambda msg: logs.append(f"Console Error: {msg.text}") if msg.type == "error" else None)

            try:
                response = await page.goto(f"{base_url}{route}", wait_until="networkidle")
                if not response:
                    print(f"Failed to load {route}: No response")
                    continue

                print(f"Status: {response.status}")

                if errors:
                    print("Runtime Errors:")
                    for e in errors:
                        print(f"  - {e}")

                if logs:
                    print("Console Errors:")
                    for l in logs:
                        print(f"  - {l}")

                if not errors and not logs and response.status < 400:
                    print(f"Route {route} loaded successfully with no visible errors.")
                elif response.status >= 400:
                    print(f"Route {route} returned status {response.status}")

            except Exception as e:
                print(f"Failed to navigate to {route}: {e}")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(run())

File: smoke-test.ts
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  const routes = ['/', '/apply', '/auth', '/redesign', '/track'];
  const baseUrl = 'http://localhost:8080';

  for (const route of routes) {
    console.log(`\n--- Testing route: ${route} ---`);
    const errors: string[] = [];
    const logs: string[] = [];

    page.on('pageerror', (err) => {
      errors.push(`Page Error: ${err.message}`);
    });

    page.on('console', (msg) => {
      if (msg.type() === 'error') {
        logs.push(`Console Error: ${msg.text()}`);
      }
    });

    try {
      const response = await page.goto(`${baseUrl}${route}`, { waitUntil: 'networkidle' });
      if (!response) {
        console.log(`Failed to load ${route}: No response`);
        continue;
      }

      console.log(`Status: ${response.status()}`);

      if (errors.length > 0) {
        console.log('Runtime Errors:');
        errors.forEach(e => console.log(`  - ${e}`));
      }

      if (logs.length > 0) {
        console.log('Console Errors:');
        logs.forEach(l => console.log(`  - ${l}`));
      }

      if (errors.length === 0 && logs.length === 0 && response.status() < 400) {
        console.log(`Route ${route} loaded successfully with no visible errors.`);
      }

    } catch (err) {
      console.log(`Failed to navigate to ${route}: ${(err as Error).message}`);
    }
  }

  await browser.close();
})();
